docs/docs/themes/api-reference.md

```diff
@@ -0,0 +1,126 @@
+---
+title: Themes API Reference
+---
+
+> ⚠⚠ Gatsby Themes are currently experimental ⚠⚠
+
+## Table of contents
+
+- [Core Gatsby APIs](#core-gatsby-apis)
+- [Configuration](#configuration)
+- [Component shadowing](#component-shadowing)
+- [Theme composition](#theme-composition)
+- [Add theme transpilation](#add-theme-transpilation)
+
+## Core Gatsby APIs
+
+Themes are packaged Gatsby sites, so you have access to all of Gatsby's APIs.
+
+- [Gatsby Config](https://www.gatsbyjs.org/docs/gatsby-config/)
+- [Actions](https://www.gatsbyjs.org/docs/actions/)
+- [Node Interface](https://www.gatsbyjs.org/docs/node-interface/)
+- ... [and more](https://www.gatsbyjs.org/docs/api-specification/)
+
+If you're new to Gatsby you can get started by following along with the guides for building out a site. Converting it to a theme will be straightforward later on since themes are prepackaged Gatsby sites.
+
+## Configuration
+
+Similarly to plugins, you can access options that are passed to your theme.
+You can use this to allow make filesystem sourcing configurable, accepting different nav menu items, or change branding colors from the default.
+
+```js:title=gatsby-config.js
+module.exports = {
+  __experimentalThemes: [
+    {
+      resolve: "gatsby-theme-name",
+      options: {
+        postsPath: "/blog",
+        colors: {
+          primary: "tomato",
+        },
+      },
+    },
+  ],
+}
+```
+
+In your theme's `gatsby-config.js` you can return a function, the argument it receives are the options:
+
+```js:title=gatsby-config.js
+module.exports = themeOptions => {
+  console.log(themeOptions)
+
+  return {
+    plugins: [
+      // ...
+    ],
+  }
+}
+```
+
+Then, in your theme's `gatsby-node.js` you can access them as the second argument:
+
+```js:title=gatsby-node.js
+exports.createPages = async ({ graphql, actions }, themeOptions) => {
+  console.log(themeOptions)
+}
+```
+
+## Component Shadowing
+
+Gatsby Themes allow you to customize any file in a theme's `src` directory by following a file naming convention.
+If you're using `gatsby-theme-tomato` which uses a `ProfileCard` component located at `src/components/ProfileCard.js` you can override the component by creating `src/gatsby-theme-tomato/components/profile-card.js`. If you want to see what props are passed you can do so by putting the props into a `pre` tag:
```

Comment:
Good point, yeah we should probably use kebab case for file names since that's Gatsby convention. I'll run through and make sure that it's reflected everywhere.